### PR TITLE
Clarify .well-known response describes the sites compliance for the session

### PR DESCRIPTION
**Pull request by pes10k**
This PR tries to clarify that the .well-known response is describing the site's respect for the GPC signal w/r/t the requesting user, and the requesting user's browsing session.

The .well-known response is not intended to describe what the site does for all users, for all sessions (though of course the site could provide the same response to all users, indicating that the site does (or does not) respect GPC equally for all users/sessions.

### CODE REVIEW

**Comment by braedon** [comment]
As the spec stands, the well-known resource *is* intended to describe what the site does for all users (where a relevant privacy law has jurisdiction).

I can't speak to what the authors intended, but that's how it reads to me, to everyone I've talked to about it, and how it's treated in every implementation I've come across.

Without getting into what the semantics of the resource *should* be, I don't think we should fundamentally change the meaning of an existing resource/field at this stage.

**Comment by j-br0** [comment]
I think the current text is somewhat ambiguous. It says: "The purpose of a GPC Support Resource is for a site to convey its support for the Global Privacy Control." I think this could be read to mean either its support for GPC for this particular user or more broadly.

Right now, the `well-known` field only allows the site to respond with `gpc` as `true` or `false`. If it is a static resource that returns the same result for all users, it's not entirely clear what `true` means. Does it mean that it honors GPC for all users around the world, that it honors GPC in jurisdictions where it's explicitly required, or something else? And separately, there's a question of _how_ it honors GPC: does it stop third-party sharing entirely, just turn off certain advertising pixels, or does it . . . change the default font size.

Pete's PR would address the first question --- the `well-known` provided to a user would tell whether that user's GPC signals are being honored (though it still wouldn't provide information how). Others should speak to the technical challenges with such a dynamic approach, but it would certainly provide more granular and robust data about GPC compliance.

Alternatively, we could devise a static resource to provide more detail about the jurisdictions in which it supports GPC instead of the binary result returned today. Presumably companies are honoring GPC is jurisdictions where they are clearly legally required to do so, but the `well-known` could signal when companies are honoring in additional jurisdictions (including jurisdictions where it is unclear what legal effect GPC has).

Least ambitiously, we could just revise the spec to clarify (1) that the resource is static and (2) what `true` means.

I think we should at least do that last bit for clarity's sake. In general, I am leery about introducing major substantive changes to the spec given that it is widely implemented and has legal consequences in some jurisdictions, but I think we can be more flexible with regard to the optional `well-known` (and the `lastUpdated` field would signal whether the company was providing data pursuant to the `well-known`'s original parameters or in a newer format).

**Comment by michael-oneill** [comment]
If servers applied different rules to different users then it would have to examine request headers (e.g. cookies) or the user's source IP,  either of which possibly being personal data. This would not look good in a privacy motivated spec.
However a machine readable declaration of a servers intent to honour a users right to privacy is a good concept, and if jurisdiction specific information would help then this could be conveyed via additional properties in the .well-known resource.

**Comment by SebastianZimmeck** [comment]
> As the spec stands, the well-known resource is intended to describe what the site does for all users (where a relevant privacy law has jurisdiction).

Agreed with @braedon. That is also how I read it. It is the flip side of the spec. The spec intentionally does not define what legal meaning GPC has. Rather, its meaning is determined by the law the site operator is subject to. Similarly, with the .well-known resource the site operator declares "I am respecting GPC to the extent I am required to under the laws applicable to me." (The operator can also go beyond the law and apply GPC where it is not required.)

> I think we should at least do that last bit for clarity's sake. 

Also agreed with @j-br0. If it is not clear, we should clarify the meaning.

File: index.html [hunk 1]
```diff
@@ -307,8 +307,10 @@ <h2>JavaScript Property to Detect Preference</h2>
       <h2>GPC Support Resource</h2>
       <p>
         A site MAY produce a resource at a .well-known URL in order for a site to represent the fact
-        that it abides by GPC. The purpose of a GPC Support Resource is for a site to convey its
-        support for the Global Privacy Control. By default, an origin's support is <em>unknown</em>.
+        that the site is respecting the GPC signal for the user requesting the
+        resource. The purpose of a GPC Support Resource is for a site to convey
+        the site's support for the Global Privacy Control for the current
+        session. By default, an origin's support is <em>unknown</em>.
```
Comment (martinthomson) [inline index.html:313]: What does "current session" mean?  This isn't a facetious question, but a real problem.  The notion of a session has never been well defined.

For background, when it comes to HTTP resources making claims (such as whether GPC is respected), they can only make claims about a single resource.  A well-known resource allows a resource to make statements for an origin.  That's a useful notion here, but if the goal is to narrow the scope of these claims based on other information (session, client jurisdiction, whatever), then the way in which that is narrowed needs to be carefully defined.  For one, the implications for HTTP caches start to look very interesting when you do that.

Answering the questions @j-br0 asks might be better way to start digging into this question of scope.

**Comment by pes10k** [comment]
If it'd help to define the concept "current session" as its used in the proposal, im happy to do so. But if whats being asked for is to define a technical, machine enforceable concept for what counts as the "current session", i do not think thats necessary or w/in the scope of this spec. Whats meant by "session" for "session cookies" is similarly ambiguous and left up to the [browser implementors](https://datatracker.ietf.org/doc/html/rfc6265#section-4.1.2) and sites (e.g., "your session has ended, please log in again") to work out.

I'm not saying it wouldn't be useful to the platform to have such a definition, im only saying i. this proposal isn't the only place that references the concept of a session without providing a complete definition of it, and ii. i dont think this spec needs to be clearer than other parts of the platform for it to still be useful.

I think we need something like this though, and to make it clear that .well-known doesn't describe a site wide policy.  Even if we extended the .well-known to include additional information, like "these are the jurisdictions or regions or laws I respect GPC under", it still wouldn't be sufficient to answer the main (to my mind) question of "if I interact with the site now, will the site respect my GPC-relevant rights."

For example, if we go the other direction, and have the .well-known describe the site's policy. We could go even further and 
extend the .well-known to  say "I respect CCPA" or something like that. But that doesn't tell the user (who may be connected through Tor, or a VPN, or living in California but on vacation in Chicago, or vise versa) if the site thinks the user is entitled to, and so the site will follow, CCPA when the user interacts with the site.

Last note, like i mentioned on the calls, this isn't a core thing to me; I think the proposal would be more useful if there was a way for the site to signal back to the user what GPC-relevant rights the site is respecting in regards to the user, and so i think that requires a change at least similar to this PR. But, i think GPC would still be useful w/o that, and so having a .well-known isn't a hill I want to die on.

But, i don't think ".well-know describes the site, not the site+user interaction" is useful for anything other than a public statement of "GPC is good", and I think we could do that w/o the .well-known.

TL;DR; my pref is: "`.well-known` describes the session" > nothing > "`.well-known` describes the site

**Comment by j-br0** [comment]
In a perfect world, I would definitely want the response to be session-specific. My three concerns are:

(1) It may be technically challenging in some cases (as @AramZS and @dmarti have argued).
(2) As @braedon notes, tens of thousands of sites have already published a `well-known` for GPC assuming a different meaning.
(3) Publishing a `well-known` is voluntary, and I fear sites will not be sufficiently incentivized to provide a session-specific response. In addition to any coding challenges, doing so potentially invites liability if they provide an inadvertently incorrect response, and I am not aware of any user agents that are taking steps to treat publishers differently based on the `well-known` (though that could change if the `well-known` was dynamic).

I am not sure these three concerns are insurmountable, just noting them. In fact, it would be my preference to surmount them.

I think I disagree that nothing > a static `well-known` however. While honoring GPC is mandatory in California and Bermuda (and possibly other jurisdictions), I think there is still some lack of awareness and/or compliance with GPC. At IAPP yesterday on the CA regulators' panel, the speakers noted that compliance with CCPA more broadly was very inconsistent. Allowing publishers to express _any_ recognition of GPC is still valuable and provides data to researchers and regulators; the trend lines detected by the GPC Support Bot which crawls `well-knowns` have been useful.

So I would still support a static resource over nothing. But if we go that route we should revise the spec to clarify whether `true` means:

(1) "I do something, somewhere in response to GPC"
(2) "I do something, everywhere in response to GPC" (but I think this would be a significant change from how many companies are interpreting this today)
(3) "I do something in response to GPC in the places where I am legally required to"
(4) "I do something in response to GPC in these jurisdictions" (and we provide guidance on how to list them out), or
(5) something else

(also, to be clear, as with @pes10k, none of this is a hill I am willing to die on)

**Comment by michael-oneill** [comment]
It would be very helpful to include information on consent-exempt storage aka "strictly necessary" cookies etc.
In Europe a website does not require consent for particular kinds of storage such as those that are "strictly necessary for the provision of an information society service requested by the subscriber or user" (ICO guidance).
The GPC signal is a No-Op in Europe (because ePrivacy already establishes default opted-out) but incorporating the ability to declare autonomy relevent information in a machine-readable way could make it _de rigueur_

**Comment by darobin** [comment]
Is there a reason not to have two fields? I don't think that JSON is running out. Like:
- `gpc: true`: this site intends to honour GPC requests as applicable.
- `requestGPCState`: is `true` if the _current request_ would honour a GPC signal that if set, `false` if it wouldn't, and undefined if it's not saying. (Let's not talk about sessions, only lawyers think that those exist.)

Side note: @michael-oneill GPC is entirely unrelated to cookies, by design. It's not that it's a no-op, it's that it deliberately targets the sale of data, not cookies considerations. The spec states so.

**Comment by michael-oneill** [comment]
> Side note: @michael-oneill GPC is entirely unrelated to cookies, by design. It's not that it's a no-op, it's that it deliberately targets the sale of data, not cookies considerations. The spec states so.

Exactly. The header signal is a no-op in Europe because it has no legal meaning there.

FYI ePrivacy is not about "cookies" either, it is about tracking, or recognising an individual across multiple web interactions by utilising data stored in their UA. The "cookie law" ephthet was given it by the tracking lobby.

I was referring to extending the well-known resource - which could give the recomendation some point in Europe, where there is a need for a machine-readable way for sites to declare storage claimed to be exempt from the prior consent requirement.

**Comment by pes10k** [comment]
> Is there a reason not to have two fields? I don't think that JSON is running out. Like:

"I dont think that JSON is running out" is a very funny phrase :)

I'm all for this idea @darobin! A+, way to cut the knot!

If no one disagrees by EOD Monday, i'll update the PR accordingly

**Comment by pes10k** [comment]
one small note though, i think session is better than request here, bc its not useful to know if GPC is being respected for the request for the `.well-known`; what the user cares about is if GPC is respected for _every other_ request to the site.

I appreciate that session is a not-100%-well-defined term on the platform, but i dont think that makes "session" meaningless (as i mentioned above, session is used similarly in other standards/specs), and it captures the intent better than request.

That said, this isn't a hill i want to die on.  If there isn't a path to close/merge this PR with "session", but there is for "request", I can go along with "request"

**Comment by braedon** [comment]
Agree with @pes10k that saying "GPC us honored for this request" isn't useful/will be exploited. We'd need to very carefully/explicitly define the scope, and consider how it interacts with the various laws. Seems hard, to put it mildly?

Personally, I think that's a lot of effort/complication/risk to add to the spec for something that is hard for most sites to even implement - a dynamic response based on *something* about the user is a very different prospect to a static file.

The elegance/simplicity of the existing well-known resource is it is extremely simple to implement, and offloads all that complexity to the relevant law. I don't see a good argument for complicating it currently.

Trying to report whether the site is honoring GPC for the current user/session/request/whatever seems like something for a response header and/or JS API?

**Comment by dmarti** [comment]
I also agree with @pes10k and @braedon. A tricky problem is when GPC must be checked and used by a service provider, in 3rd-party JavaScript that may appear on multiple sites, but the `.well-known` file can only be updated in the CMS for the individual site. This is a common pattern for sites that use service providers that must (or choose to) respect GPC, and makes it hard to keep the resourceand the actual GPC behavior in sync. A site with this fairly common setup would be likely to omit the resource entirely.

**Comment by darobin** [comment]
Glad you liked it @pes10k :)

I'm not going to die on this hill either, but note that if your goal is to prevent lawyer from lying then using a somewhat vague term like "session" might not be the best idea either. Maybe make it something like "a request to any other resource on this site with similar parameters"?

**Comment by j-br0** [comment]
> Is there a reason not to have two fields? I don't think that JSON is running out. Like:
> 
> * `gpc: true`: this site intends to honour GPC requests as applicable.
> * `requestGPCState`: is `true` if the _current request_ would honour a GPC signal that if set, `false` if it wouldn't, and undefined if it's not saying. (Let's not talk about sessions, only lawyers think that those exist.)

I could live with two fields, but I suspect in practice few if any will implement the proposed `requestGPCState` field for the reasons discussed in the thread (novel, difficult to implement, potentially induces legal liability). Earlier proposed versions of the CCPA regulations would have required sites to display opt-out state to users, but there is no such mandate today under any law I am aware of. And even if that mandate is reintroduced, I'm not sure the `well-known` is the best mechanism to facilitate compliance.

Currently at least 20k sites seem to voluntarily publishing a GPC `well-known`, and the trend lines are useful to both researchers and regulators. Of the options I listed earlier to more narrowly define what a static resource is meant to convey, I would choose to define `gpc: true` as "this site honors GPC signals at least in the jurisdictions where it is legally required to do so." Such a statement would not trigger new liability for site owners which could otherwise deter implementation. While the signal would only be conveying that the sites were compliant with the law, we know today a lot of sites _aren't_ compliant with the law, so the `well-known` would provide some information to the ecosystem about awareness and adoption.

**Comment by AramZS** [comment]
Want to add this note to the record: 

From CCPA regs: 

https://cppa.ca.gov/regulations/pdf/cppa_regs.pdf#page=37

```
(6) A business may display whether it has processed the consumer’s opt-out preference
signal as a valid request to opt-out of sale/sharing on its website. For example, the
business may display on its website “Opt-Out Preference Signal Honored” when a
browser, device, or consumer using an opt-out preference signal visits the website, or
display through a toggle or radio button that the consumer has opted out of the sale
of their personal information.
```

From the CG conversation

**Comment by AramZS** [comment]
CG felt strongly that this did not particularly satisfy any concerns about the resource and may have introduced new ones as such closing without merging at this time.
